roles/variables/defaults/main.yml

```diff
@@ -566,20 +567,32 @@ kafka_broker_default_listeners: "{
     'port': 9092,
     'ssl_enabled': {% if ccloud_kafka_enabled|bool %}true{% else %}{{ssl_enabled|string|lower}}{% endif %},
     'ssl_mutual_auth_enabled': {% if ccloud_kafka_enabled|bool %}false{% else %}{{ssl_mutual_auth_enabled|string|lower}}{% endif %},
-    'sasl_protocol': '{% if rbac_enabled|bool or oauth_enabled|bool %}OAUTH{% elif ccloud_kafka_enabled|bool %}PLAIN{% else %}{{sasl_protocol}}{% endif %}'
-  }{% if kafka_broker_configure_multiple_listeners|bool %},
+    'ssl_client_authentication': '{% if ccloud_kafka_enabled|bool %}none{% else %}{{ssl_client_authentication|string|lower}}{% endif %}',
+    'sasl_protocol': '{% if auth_mode in [\"ldap\", \"ldap_with_oauth\", \"oauth\"] %}OAUTH{% elif ccloud_kafka_enabled|bool %}PLAIN{% else %}{{sasl_protocol}}{% endif %}'
+  }{% if auth_mode == 'mtls' %},
+  'internal_token': {
+    'name': 'INTERNAL_TOKEN',
+    'port': 9088,
+    'ssl_enabled': true,
+    'ssl_mutual_auth_enabled': true,
+    'ssl_client_authentication': '{{ssl_client_authentication|string|lower}}',
+    'sasl_protocol': 'OAUTH'
+  }{% endif %}
+  {% if kafka_broker_configure_multiple_listeners|bool %},
   'broker': {
     'name': 'BROKER',
     'port': 9091,
     'ssl_enabled': {{ssl_enabled|string|lower}},
     'ssl_mutual_auth_enabled': {{ssl_mutual_auth_enabled|string|lower}},
+    'ssl_client_authentication': '{{ssl_client_authentication|string|lower}}',
     'sasl_protocol': '{{sasl_protocol}}'
   }{% if kafka_broker_configure_control_plane_listener|bool and not kraft_enabled|bool %},
   'controller': {
     'name': 'CONTROLLER',
     'port': 9089,
     'ssl_enabled': {{ssl_enabled|string|lower}},
     'ssl_mutual_auth_enabled': {{ssl_mutual_auth_enabled|string|lower}},
+    'ssl_client_authentication': '{{ssl_client_authentication|string|lower}}',
```

Comment:
Test version upgrades to ensure that adding new fields in listeners dict doesnt cause trouble in upgades